### PR TITLE
[Subscriptions API] CESQL filter

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Added the `sql` filter dialect to subscriptions api filters.

### CODE REVIEW

File: subscriptions-api.md [hunk 1]
```diff
@@ -736,6 +736,19 @@ For example:
 }
 ```
 
+###### `sql` filter dialect
+
+Use of this MUST have a string value, representing a [CloudEvents SQL Expression](./expression-language.md).
+The filter result MUST be false if an error occurred while evaluating the expression or if the result value, 
+coerced to boolean, equals to the `FALSE` boolean value, otherwise the filter result MUST be true.
```
Comment (duglin) [inline subscriptions-api.md:743]: This is ok, but it just felt weird to call out the cases where it's "false" and then say "everything else is 'true' ".
Normally I would expected it to say "it's true if the expression evaluates to true, and all other cases evaluate to false".  But I guess either way works. :-)

**Comment by duglin** [comment]
LGTM

File: subscriptions-api.md (hunk 1 above)
Comment (slinkydeveloper) [reply subscriptions-api.md:743]: I wanted to be explicit about the fact that if there's an error, then the result is false. But I can swap the sentence if you want to.

Comment (duglin) [reply subscriptions-api.md:743]: we chatted about this on the call today and overall people are ok with the PR, but would prefer if you turned it around and made it more like:   if it valuates to 'true' then the result is 'true', other all other cases (including an error) results in 'false' (in your own words).

Comment (slinkydeveloper) [reply subscriptions-api.md]: done, check it out now

**Comment by duglin** [comment]
@slinkydeveloper can you rebase?

**Comment by duglin** [comment]
Approved on the June 3 call - will merge after the rebase

**Comment by slinkydeveloper** [comment]
@duglin done, ready for merging

**Comment by duglin** [comment]
Thanks - merging!

**Pull request merged**